source/sigv4.c

```diff
@@ -3291,6 +3293,9 @@ SigV4Status_t SigV4_GenerateHTTPAuthorization( const SigV4Parameters_t * pParams
      * Note that the StringToSign starts from the beginning of the processing buffer. */
     if( returnStatus == SigV4Success )
     {
+        /* MISRA Ref 18.2.1 [Pointer subtraction within array] */
+        /* More details at: https://github.com/aws/SigV4-for-AWS-IoT-embedded-sdk/blob/main/MISRA.md#rule-182 */
+        /* coverity[misra_c_2012_rule_18_2_violation] */
         bufferLen = canonicalContext.pBufCur - ( char * ) canonicalContext.pBufProcessing;
```

Comment:
Could this pBufCur's type be changed from a pointer to an array index?

Reply:
> Rule 18.2 Subtraction between pointers shall only be applied to pointers that
address elements of the same array

For the [header file](https://github.com/aws/SigV4-for-AWS-IoT-embedded-sdk/blob/dbe5583e50a6fd232f35c728c5d0941e4f012ff1/source/include/sigv4_internal.h#L209), the `pBufCur` is a `pBufProcessing` curosr and addresses `pBufProcessing` elements in the design. This array substraction follows MISRA rule 18.2.

It looks like coverity can't tell if they address elements of the same array. Therefore, I added an exception here. I also added assert here to ensure that pBufCur points to the same array.

Reply:
Yes, I am suggesting we change that. Using indexes into the buffer instead of pointers into the buffer is probably a better design, and would not trip up coverity.

Reply:
We discussed this and we shall fix the above issue by refactoring in a separate PR. That will keep this PR 'clean'.